rfcs/20241205-graph-api-causal-mask/README.md

```diff
@@ -0,0 +1,485 @@
+
+
+| Attribute Name | Description | Value Type | Supported Values | Required or Optional |
+| :------------- | :---------- | :--------- | :--------------- | :------------------- |
+| [axis] (@ref dnnl::graph::op::attr::axis) | Specifies the dimension along which index values are generated. | s64 | A s64 value in the range of [-r, r-1] where r = rank(src) | Required |
+
+##### Execution arguments
+
+##### Inputs
+
+| Index | Argument Name | Required or Optional |
+| :---- | :------------ | :------------------- |
+| 0     | `src`         | Required             |
+
+##### Outputs
+
+| Index | Argument Name | Required or Optional |
+| :---- | :------------ | :------------------- |
+| 0     | `dst`         | Required             |
+
+##### Supported data types
+
+`GenInex` operation supports the following data type combinations.
+
+| src  | dst  |
+| :--- | :--- |
+| f32  | s64  |
+| f16  | s64  |
+| bf16 | s64  |
+
+#### GreaterThan Op Support
+
+`GreaterThan` performs element-wise comparison with two given tensors applying
+broadcast rules specified in the `auto_broadcast` attribute.
+
+| Attribute Name | Description | Value Type | Supported Values | Required or Optional |
+| :------------- | :---------- | :--------- | :--------------- | :------------------- |
+| [auto_broadcast](@ref dnnl::graph::op::attr::auto_broadcast) | Specifies rules used for auto-broadcasting of src tensors. | string     | `none`, `numpy` (default) | Optional             |
+
+##### Inputs
+
+| Index | Argument Name | Required or Optional |
+| :---- | :------------ | :------------------- |
+| 0     | `src_0`       | Required             |
+| 1     | `src_1`       | Required             |
+
+Both src shapes should match if `auto_broadcast` attributes is `none`. Otherwise
+if `auto_broadcast` attributes is `numpy`, `src_0` and `src_1` can have different
+shapes and auto-broadcasting is allowed.
+
+##### Outputs
+
+| Index | Argument Name | Required or Optional |
+| :---- | :------------ | :------------------- |
+| 0     | `dst`         | Required             |
+
+##### Supported data types
+
+GreaterThan operation supports the following data type combinations.
+
+| src_0 / src_1  | dst     |
+| :------------- | :------ |
+| f32            | boolean |
+| f16            | boolean |
+| bf16           | boolean |
+
+#### Constant Op Support
+
+`Constant` operation takes a scalar as an attribute, and converts it to a tensor,
+which will then be passed to *Add* and *Sub* operations in the case of bottom-right
+causal mask to provide the information of s_q and s_kv. These values are needed during
+compilation stage to generate optimized kernels.
+
+| Attribute Name | Description | Value Type | Supported Values | Required or Optional |
+| :------------- | :---------- | :--------- | :--------------- | :------------------- |
+| [scalar] (@ref dnnl::graph::op::attr::scalar) | Specifies the constant value. | f32, bf16, f16 | arbitrary valid floating point value | Required             |
```

Comment:
not sure if bf16 or f16 can work for bottom-right, especially for bf16, only 7 mantissa bits. it can't be used to represent the exact integer number due to the rounding error. maybe we need to add some pattern check when used for mask pattern